### ISSUE TITLE
libbpf error in calico-node's log after update to v3.21

### ISSUE THREAD

**Issue by ml14tlc** [projectcalico/calico#5381]
## Expected Behavior
The logs of the `calico-node` pods are clean

## Current Behavior
the `calico-node logs` are filled with the following message

```
2022-01-04 09:19:42.923 [WARNING][56781] felix/bpf_ep_mgr.go 608: Failed to apply policy to interface, will retry error=error loading program error loading object operation not permitted
2022-01-04 09:19:53.209 [INFO][56781] felix/bpf_ep_mgr.go 1339: Need to attach program hook=“egress” iface=“vxlan.calico” type=“host”
2022-01-04 09:19:53.209 [INFO][56781] felix/bpf_ep_mgr.go 1339: Need to attach program hook=“ingress” iface=“vxlan.calico” type=“host”
libbpf: couldn’t retrieve pinned map ‘/sys/fs/bpf/tc/vxlan.calico_igr/cali_jump’: Operation not permitted
libbpf: map ‘cali_jump’: error reusing pinned map
libbpf: map ‘cali_jump’: failed to create: Operation not permitted(-1)
libbpf: failed to load object ‘/tmp/calico-tc802504830/from_hep_fib_no_log.o’
libbpf: couldn’t retrieve pinned map ‘/sys/fs/bpf/tc/vxlan.calico_egr/cali_jump’: Operation not permitted
libbpf: map ‘cali_jump’: error reusing pinned map
libbpf: map ‘cali_jump’: failed to create: Operation not permitted(-1)
libbpf: failed to load object ‘/tmp/calico-tc805590747/to_hep_no_log.o’
```

## Possible Solution
As suggested in the [forum](https://discuss.projectcalico.org/t/libbpf-error-after-update-to-v3-21/366/2), it looks like a bug

## Steps to Reproduce (for bugs)
This is how I stumbled into this

1. Install Calico v3.20.1 using the manifest, according to the [instructions](https://projectcalico.docs.tigera.io/getting-started/kubernetes/self-managed-onprem/onpremises#install-calico-with-kubernetes-api-datastore-50-nodes-or-less) for 50 nodes or less, with K8s API store
2. Configure it to use VXLAN and eBPF
3. [migrate from manifest to operator](https://projectcalico.docs.tigera.io/maintenance/operator-migration), while upgrade to v3.21

## Context
My set up is rather basic, I'm not using network policies yet. Pods can communicate, but the (repeated) error message seems to imply that somethig's going wrong

## Your Environment

* Calico version: 3.21
* Orchestrator version (e.g. kubernetes, mesos, rkt): Kubernetes 1.21.3
* Operating System and version: Ubuntu 20.04 with kernel `5.4.0-91-generic #102-Ubuntu SMP Fri Nov 5 16:31:28 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux` (also tried `5.8.0-63-generic #71~20.04.1-Ubuntu SMP` but it didn’t help)
* Link to your project (optional):

**Comment by tomastigera** [comment]
cc @sridhartigera - i think it was already addressed, right?

**Comment by sridhartigera** [comment]
@tomastigera yes. The fix is in v3.21

**Comment by caseydavenport** [comment]
We'll be cutting a v3.21 patch release early next week if all goes to plan. Stay tuned!
